crates/weaver_resolver/src/lib.rs

```diff
@@ -316,40 +301,36 @@ impl SchemaResolver {
                                     let relative_path = &path[prefix.len() + 1..];
                                     format!("{}/{}", registry_path_repr, relative_path)
                                 };
-                                vec![Ok((path, spec))].into_par_iter()
-                            }
-                            Err(e) => match e {
-                                weaver_semconv::Error::CompoundError(errors) => errors
-                                    .into_iter()
-                                    .map(|e| Err(Error::SemConvError { error: e }))
-                                    .collect::<Vec<_>>()
-                                    .into_par_iter(),
-                                _ => vec![Err(Error::SemConvError { error: e })].into_par_iter(),
+                                (path, spec)
                             },
-                        }
+                        )]
+                        .into_par_iter()
                     }
-                    Err(e) => vec![Err(Error::FailToWalkDir {
+                    Err(e) => vec![WResult::FatalErr(weaver_semconv::Error::SemConvSpecError {
                         error: e.to_string(),
                     })]
                     .into_par_iter(),
                 }
             })
             .collect::<Vec<_>>();
 
-        let mut error = vec![];
-        let result = result
-            .into_iter()
-            .filter_map(|r| match r {
-                Ok(r) => Some(r),
-                Err(e) => {
-                    error.push(e);
-                    None
+        // Process all the results of the previous parallel processing.
```

Comment:
NIt: We should turn this into a helper method eventually.

Reply:
I suggest implementing this helper once we encounter more than one instance of this pattern.